mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte

```diff
@@ -26,17 +27,19 @@
 
   async function handleDeleteRecords() {
     if (!isDeleting) {
-      try {
-        isDeleting = true;
-        selection.freezeSelection = true;
-        await recordsData.deleteSelected(selectedRowIndices);
-        selection.resetSelection();
-      } catch (e) {
-        toast.fromError(e);
-      } finally {
-        selection.freezeSelection = false;
-        isDeleting = true;
-      }
+      isDeleting = true;
+      selection.freezeSelection = true;
+      void confirmDelete({
+        identifierType: 'Row',
+        onProceed: () => recordsData.deleteSelected(selectedRowIndices),
+        onError: (e) => toast.fromError(e),
+        onSuccess: () =>
+          toast.success({
+            title: 'Row deleted successfully!',
+          }),
+      });
+      selection.freezeSelection = false;
```

Comment:
There is no need of freezing the selection now since the modal will take care of that.

Reply:
sure removed it !